src/services/file.service.ts

```diff
@@ -115,9 +119,50 @@ export class FileService {
 
     this.logger.silly('Upload file to Jupiter');
     try {
-      return await uploader.writeFile(file.filename, file.buffer);
+      if (!userInfo.password){
+        throw new Error('[uploadFileWithJupiterFs]: Password needs to be set');
+      }
+      const buffer = this.encryptFile(file.buffer, userInfo.password, this.algorithm);
+      return await uploader.writeFile(file.filename, buffer);
     } catch (error) {
+      this.logger.error('[Write File]:' + JSON.stringify(error))
       throw JupiterError.parseJupiterResponseError(error);
     }
   }
+
+  /**
+   * Encrypt the buffer file
+   * @param buffer
+   * @param password
+   * @param algorithm
+   */
+  public encryptFile(buffer: Buffer, password: string, algorithm: string = 'aes-256-ctr'){
```

Comment:
please use aes-256-cbc
also it might be better not to have a default.
also the algorithm should come from env vars

Reply:
done